libs/domain/search/facet-range/facet-range.component.ts

```diff
@@ -0,0 +1,153 @@
+import { RangeFacet, RangeFacetValue } from '@spryker-oryx/product';
+import { MultiRangeChangeEvent } from '@spryker-oryx/ui/multi-range';
+import {
+  computed,
+  debounce,
+  effect,
+  elementEffect,
+  signalProperty,
+} from '@spryker-oryx/utilities';
+import { TemplateResult, html } from 'lit';
+import { state } from 'lit/decorators.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+// use relative path for dev ssr server, SearchFacetComponent is undefined
+import { SearchFacetComponent } from '../facet/facet.component';
+import {
+  SearchFacetRangeComponentAttributes,
+  SearchFacetRangeComponentValues,
+} from './facet-range.model';
+import { searchRangeFacetStyles } from './facet-range.styles';
+
+export class SearchRangeFacetComponent
+  extends SearchFacetComponent
+  implements
+    SearchFacetRangeComponentAttributes,
+    SearchFacetRangeComponentValues
+{
+  static styles = [...SearchFacetComponent.styles, searchRangeFacetStyles];
+
+  @signalProperty({ type: Number }) step = 1;
+  @signalProperty() labelMin?: string;
+  @signalProperty() labelMax?: string;
+  @state() min?: number;
+  @state() max?: number;
+
+  protected $isDirty = computed(() => {
+    const facet = this.facet() as RangeFacet;
+
+    if (!facet) return false;
+
+    const {
+      values: { min, max, selected },
+    } = facet;
+
+    return selected?.min !== min || selected?.max !== max;
+  });
+
+  @elementEffect()
+  protected $syncValues = effect(() => {
+    const facet = this.facet() as RangeFacet;
+
+    if (!facet) return;
+
+    const {
+      values: { min, max, selected },
+    } = facet;
+
+    this.min = selected?.min ?? min;
+    this.max = selected?.max ?? max;
+  });
+
+  protected onRangeChange = debounce(
+    (e: CustomEvent<MultiRangeChangeEvent>): void => {
+      const { minValue: min, maxValue: max } = e.detail;
+      const selected = { min, max };
+
+      if (this.hasChangedValue(selected)) {
+        this.min = min;
+        this.max = max;
+        this.controller.dispatchSelectEvent({ selected });
+      }
+    },
+    300
+  );
+
+  protected onBlur(e: InputEvent): void {
+    const { value, name } = e.target as HTMLInputElement;
+    this[name as keyof SearchFacetRangeComponentValues] = Number(value);
+  }
```

Comment:
But how it's possible to change value and has not been notified about it before blur?

Reply:
According the criteria for the story